### PR TITLE
PHPLIB-1582: Support hint option for distinct command

### PR DESCRIPTION
**Pull request by jmikola**
https://jira.mongodb.org/browse/PHPLIB-1582

### CODE REVIEW

File: src/Operation/Distinct.php [hunk 1]
```diff
@@ -175,6 +186,11 @@ private function createCommandDocument(): array
             $cmd['collation'] = (object) $this->options['collation'];
         }
 
+        if (isset($this->options['hint'])) {
+            /** @psalm-var string|array|object */
```
Comment (GromNaN) [inline src/Operation/Distinct.php]: Do you really need this annotation? If it applies to `$cmd['hint']`, it should be `string|object`, without `array`

File: src/Operation/Distinct.php [hunk 2]
```diff
@@ -83,6 +90,10 @@ public function __construct(private string $databaseName, private string $collec
             throw InvalidArgumentException::expectedDocumentType('"collation" option', $this->options['collation']);
         }
 
+        if (isset($this->options['hint']) && ! is_string($this->options['hint']) && ! is_array($this->options['hint']) && ! is_object($this->options['hint'])) {
```
Comment (GromNaN) [inline src/Operation/Distinct.php]: The doc says `string|document`, the logical test should use `is_document`.

```suggestion
        if (isset($this->options['hint']) && ! is_string($this->options['hint']) && ! is_document($this->options['hint'])) {
```

File: src/Operation/Distinct.php [hunk 3]
```diff
@@ -175,6 +186,11 @@ private function createCommandDocument(): array
             $cmd['collation'] = (object) $this->options['collation'];
         }
 
+        if (isset($this->options['hint'])) {
+            /** @psalm-var string|array|object */
+            $cmd['hint'] = is_array($this->options['hint']) ? (object) $this->options['hint'] : $this->options['hint'];
```
Comment (GromNaN) [inline src/Operation/Distinct.php:191]: Isn't the `array` automatically converted to a BSON document? Or maybe that handles edge cases like `[]` or `[0 => 'foo']`.

File: tests/Operation/DistinctTest.php [hunk 4]
```diff
@@ -42,6 +43,7 @@ public function testExplainableCommandDocument(): void
     {
         $options = [
             'collation' => ['locale' => 'fr'],
+            'hint' => '_id_',
```
Comment (GromNaN) [inline tests/Operation/DistinctTest.php:46]: The `hint` option is supported in `explain` command on server version < 7.1.0?

Comment (jmikola) [reply tests/Operation/DistinctTest.php:46]: I assume not. This is just a unit test, so we're only asserting that the option is included via the Explainable interface.

File: src/Operation/Distinct.php (hunk 3 above)
Comment (jmikola) [reply src/Operation/Distinct.php:191]: This is just for the edge case you highlighted.

File: src/Operation/Distinct.php (hunk 1 above)
Comment (jmikola) [reply src/Operation/Distinct.php]: Psalm did complain with out this. Good catch, though, as I can remove `array` here.

File: src/Operation/Distinct.php (hunk 2 above)
Comment (jmikola) [reply src/Operation/Distinct.php]: Good catch. I'll also make a separate ticket to migrate other `hint` option validation to use `is_document`, as it looks like we missed that when introducing PackedArray.

Comment (jmikola) [reply src/Operation/Distinct.php]: Opened [PHPLIB-1587](https://jira.mongodb.org/browse/PHPLIB-1587) to track that.

**Review by GromNaN — APPROVED** [review]
